File: BACK/traffic-police-go/model/VehicleDTO.go
package model

import (
	"encoding/json"
	"io"
)

type VehicleDTO struct {
	Id           string `json:"id"`
	Mark         string `json:"mark"`
	Model        string `json:"model"`
	Registration string `json:"registration"`
	Year         int    `json:"year"`
	Color        string `json:"color"`
	IsStolen     bool   `json:"isStolen"`
	OwnerId      string `json:"ownerId"`
}

type Vehicles []*VehicleDTO

func (v *VehicleDTO) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(v)
}

func (v *Vehicles) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(v)
}

func (v *Vehicles) FromJSON(r io.Reader) error {
	d := json.NewDecoder(r)
	d.DisallowUnknownFields()
	return d.Decode(v)
}

func (v *VehicleDTO) FromJSON(r io.Reader) error {
	d := json.NewDecoder(r)
	d.DisallowUnknownFields()
	return d.Decode(v)
}

File: BACK/traffic-police/src/main/java/com/example/traffic_police/model/VehicleDTO.java
package com.example.traffic_police.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class VehicleDTO {
    private String id;
    private String mark;
    private String model;
    private String registration;
    private int year;
    private String color;
    private boolean isStolen;
    private String ownerId;
}